diff --git a/SimGeneral/MixingModule/python/trackingTruthProducerSelection_cfi.py b/SimGeneral/MixingModule/python/trackingTruthProducerSelection_cfi.py
index fc379c3..da63b18 100644
--- a/SimGeneral/MixingModule/python/trackingTruthProducerSelection_cfi.py
+++ b/SimGeneral/MixingModule/python/trackingTruthProducerSelection_cfi.py
@@ -4,12 +4,12 @@ from SimGeneral.MixingModule.trackingTruthProducer_cfi import trackingParticles
 trackingParticles.select = cms.PSet(
     lipTP = cms.double(1000),
     chargedOnlyTP = cms.bool(True),
+    stableOnlyTP = cms.bool(False),
     pdgIdTP = cms.vint32(),
-    signalOnlyTP = cms.bool(True),
+    signalOnlyTP = cms.bool(False),
     minRapidityTP = cms.double(-5.0),
-    minHitTP = cms.int32(3),
+    minHitTP = cms.int32(0),
     ptMinTP = cms.double(0.1),
     maxRapidityTP = cms.double(5.0),
-    tipTP = cms.double(1000),
-    stableOnlyTP = cms.bool(False)
+    tipTP = cms.double(1000)
 )
