diff --git a/Configuration/Applications/python/ConfigBuilder.py b/Configuration/Applications/python/ConfigBuilder.py
index c393e45..0b71228 100644
--- a/Configuration/Applications/python/ConfigBuilder.py
+++ b/Configuration/Applications/python/ConfigBuilder.py
@@ -742,6 +742,7 @@ class ConfigBuilder(object):
 
         custOpt=self._options.customisation_file.split(",")
 	custMap={}
+	fileNames = []
 	for opt in custOpt:
 		if opt=='': continue
 		if opt.count('.')>1:
@@ -756,6 +757,7 @@ class ConfigBuilder(object):
 			custMap[fileName].extend(rest.split('+'))
 		else:
 			custMap[fileName]=rest.split('+')
+			fileNames.append(fileName)
 		
         if len(custMap)==0:
                 final_snippet='\n'
@@ -769,7 +771,7 @@ class ConfigBuilder(object):
 		if allFcn.count(fcn)!=1:
 			raise Exception("cannot specify twice "+fcn+" as a customisation method") 
 
-	for f in custMap:
+	for f in fileNames:
 		# let python search for that package and do syntax checking at the same time
                 packageName = f.replace(".py","").replace("/",".")
                 __import__(packageName)
